p2p/peer.py

```diff
@@ -695,28 +700,33 @@ def unsubscribe(self, subscriber: PeerSubscriber) -> None:
             peer.remove_subscriber(subscriber)
 
     async def start_peer(self, peer: BasePeer) -> None:
+        asyncio.ensure_future(peer.run())
+        await self.wait(peer.events.started.wait(), timeout=1)
         try:
             # Although connect() may seem like a more appropriate place to perform the DAO fork
             # check, we do it here because we want to perform it for incoming peer connections as
             # well.
-            msgs = await self.ensure_same_side_on_dao_fork(peer)
+            with peer.collect_sub_proto_messages() as buffer:
+                await self.ensure_same_side_on_dao_fork(peer)
```

Comment:
I know it's unrelated to this PR and I remember silently reading a discussion around this a while back so please forgive if my comment doesn't make sense but it bothers me (probably same as everyone else) that we have this dao check built right into the `PeerPool`. Did we consider to allow passing something like a `precondition_predicate` (name tbd) to the `PeerPool` that the `PeerPool` uses to check whether peers qualify for further communication. That way, we don't hardcode the dao check here and keep things flexible for other chains.

Reply:
@cburgdorf one of the tasks that I don't think there is a detailed issue for but is likely to be a core part of one of the subsequent milestones is to tease out the rest of this EVM logic from the `p2p` module.  Exactly how isn't clear but it might look something like you mentioned.